libnopegl/src/backends/vk/gpu_ctx_vk.c

```diff
@@ -66,26 +66,30 @@ static VkResult create_dummy_texture(struct gpu_ctx *s)
                    NGLI_TEXTURE_USAGE_TRANSFER_DST_BIT,
     };
 
-    VkResult res = ngli_texture_vk_init(s_priv->dummy_texture, &params);
-    if (res != VK_SUCCESS)
-        return res;
+    int ret = ngli_texture_init(s_priv->dummy_texture, &params);
+    if (ret < 0)
+        return VK_ERROR_UNKNOWN;
```

Comment:
Afaict everything except the swapchain could work with ngl retcode directly with no loss of information. Can be done later™.

Reply:
I think I'll do that in a separate PR with more cleanups.